pkg/maps/nat/per_cluster_nat.go

```diff
@@ -15,49 +14,86 @@ import (
 	"github.com/cilium/cilium/pkg/bpf"
 	cmtypes "github.com/cilium/cilium/pkg/clustermesh/types"
 	"github.com/cilium/cilium/pkg/lock"
-	"github.com/cilium/cilium/pkg/logging/logfields"
 )
 
 const (
-	PerClusterNATOuterMapPrefix     = "cilium_per_cluster_snat_"
-	perClusterNATIPv4OuterMapSuffix = "v4_external"
-	perClusterNATIPv6OuterMapSuffix = "v6_external"
-	perClusterNATMapMaxEntries      = cmtypes.ClusterIDMax + 1
+	perClusterOuterMapPrefix = "cilium_per_cluster_snat_"
+	perClusterIPv4OuterMap   = perClusterOuterMapPrefix + "v4_external"
+	perClusterIPv6OuterMap   = perClusterOuterMapPrefix + "v6_external"
+	perClusterMapMaxEntries  = cmtypes.ClusterIDMax + 1
 )
 
-// Global interface to interact with IPv4 and v6 NAT maps. We can choose the
-// implementation of this at startup time by choosing InitPerClusterNATMaps
-// or InitDummyPerClusterNATMaps for initialization.
-var PerClusterNATMaps PerClusterNATMapper
+// ClusterOuterMapName returns the name of the outer per-cluster NAT map
+// for the given IP family. It can be overwritten for testing purposes.
+var ClusterOuterMapName = clusterOuterMapName
 
-// An interface to interact with the global map.
+func clusterOuterMapName(family IPFamily) string {
```

Comment:
Looks a bit cumbersome doing the same thing over and over aka if ipv4 else ...  and xxxV4() xxxV6() parity with for the most part very similar logic?? 
Could be out of scope for this PR but seems leveraging generics would help cleaning this further??

Reply:
Yeah, I agree that there's room for improvement there, especially to reduce code duplication. I would postpone that refactoring though, as this PR is already pretty large, and most occurrences are in the `nat.go` file, which is only touched tangentially by this PR (to replace the ipv4 boolean with the IPFamily constants).